coderd/provisionerjobs.go

```diff
@@ -343,3 +338,46 @@ func convertProvisionerJob(provisionerJob database.ProvisionerJob) codersdk.Prov
 func provisionerJobLogsChannel(jobID uuid.UUID) string {
 	return fmt.Sprintf("provisioner-log-logs:%s", jobID)
 }
+
+// provisionerJobLogsMessage is the message type published on the provisionerJobLogsChannel() channel
+type provisionerJobLogsMessage struct {
+	EndOfLogs bool                         `json:"end_of_logs,omitempty"`
+	Logs      []database.ProvisionerJobLog `json:"logs,omitempty"`
+}
+
+func (api *API) followLogs(jobID uuid.UUID) (chan database.ProvisionerJobLog, func(), error) {
+	bufferedLogs := make(chan database.ProvisionerJobLog, 128)
+	closeSubscribe, err := api.Pubsub.Subscribe(provisionerJobLogsChannel(jobID),
+		func(ctx context.Context, message []byte) {
+			jlMsg := provisionerJobLogsMessage{}
+			err := json.Unmarshal(message, &jlMsg)
+			if err != nil {
+				api.Logger.Warn(ctx,
+					fmt.Sprintf("invalid provisioner job log on channel %q: %s",
+						provisionerJobLogsChannel(jobID),
+						err.Error()))
```

Comment:
Is there a reason we aren't using the structured fields?
We probably should have a metter method to ensure things like `job_id` are consistent everywhere we use it.

```suggestion
				api.Logger.Warn(ctx, "invalid provisioner job log on channel",
					slog.Error(err),
					slog.F("job_id", jobID),
				)
```

Reply:
That's a good point.  This code was just moved, so I didn't really look at it very critically.  Happy to fix up.